core/src/main/java/org/infinispan/persistence/sifs/Index.java

```diff
@@ -341,12 +354,23 @@ public CompletionStage<Void> clear() {
             processor.onNext(clearRequest);
             stage.dependsOn(clearRequest);
          }
-         for (int i = 0; i < sizePerSegment.length(); ++i) {
-            sizePerSegment.set(i, 0);
-         }
-         return stage.freeze();
-      } finally {
-         lock.writeLock().unlock();
+
+         return stage.freeze().whenComplete((ignore, t) -> {
+            lock.unlockWrite(writeStampToUnlock);
+            if (t != null) {
+               log.debugf(t, "Clear encountered an exception, size stats for future invocations may be incorrect");
```

Comment:
Should this be `errorf`?

Reply:
I would say WARN at most. I can change it to that.